tensorflow/compiler/tf2tensorrt/convert/ops/resize.cc

```diff
@@ -0,0 +1,658 @@
+      }
+    } else {
+      // Output size is given as a tensor, possibly as the result of shape
+      // calculation ops in the graph.
+      if (params_->use_implicit_batch) {
+        return errors::Unimplemented(
+            "Resize requires constant output size in implicit batch mode, at ",
+            node_def.name());
+      }
+
+      // Check that this is a shape tensor and it has two values.
+      ITensorProxyPtr size_param = inputs.at(1).tensor();
+      TRT_ENSURE(size_param->getType() == nvinfer1::DataType::kINT32);
+      TRT_ENSURE(size_param->getDimensions().nbDims == 1);
+      if (size_param->getDimensions().d[0] != 2) {
+        return errors::Unimplemented(
+            "Resize requires a 2D value for the size, at ", node_def.name());
+      }
+    }
+
+    // Verify and consume node attributes.
+    StatusOr<bool> align_corners = GetAttrValue<bool>("align_corners");
+    TRT_ENSURE_OK(align_corners);
+    attrs_.align_corners = *align_corners;
+    StatusOr<bool> half_pixel_centers =
+        GetAttrValue<bool>("half_pixel_centers");
+    TRT_ENSURE_OK(half_pixel_centers);
+    attrs_.half_pixel_centers = *half_pixel_centers;
+
+    if (node_def.op() == "ResizeBilinear") {
+      attrs_.mode = nvinfer1::ResizeMode::kLINEAR;
+    } else if (node_def.op() == "ResizeNearestNeighbor") {
+      attrs_.mode = nvinfer1::ResizeMode::kNEAREST;
+    } else {
+      return errors::Unimplemented(node_def.op(), " is not yet implemented at ",
+                                   node_def.name());
+    }
+
+    // If we require workaround and have dynamic spatial shape, this is not
+    // supported.
+    if (!IS_TRT_VERSION_GE(8, 0, 0, 0)) {
+      if (!attrs_.align_corners &&
+          !(attrs_.has_static_input_shape && attrs_.has_static_output_shape) &&
+          attrs_.mode == nvinfer1::ResizeMode::kLINEAR) {
+        VLOG(0) << "Resize parameter combination of (mode=bilinear, align ";
+        return errors::Unimplemented(
+            "Resize parameter combination of (mode=bilinear, align "
+            "corners=False, dynamic spatial shape) is unsupported.");
+      }
+    }
+    return Status::OK();
+  }
+
+  static bool UseConvResizeWorkaround(const ResizeAttributes& attrs) {
+    if (!attrs.align_corners && attrs.mode == nvinfer1::ResizeMode::kLINEAR) {
+      if (!IS_TRT_VERSION_GE(8, 0, 0, 0)) {
+        return true;
+      }
+
+      int64_t force_bilinear_resize_war = 0;
+      ReadInt64FromEnvVar("TF_TRT_FORCE_BILINEAR_RESIZE_WAR", 0,
+                          &force_bilinear_resize_war);
+      if (force_bilinear_resize_war > 0) {
+        LOG(WARNING) << "Forcing the use of LHS-dilated convolution WAR for "
+                        "ResizeBilinear when aligned_corners=false";
+        return true;
+      }
+    }
+    return false;
+  }
+
+  Status Convert() {
+    // Transpose tensor from NHWC to NCHW format. Bilinear resize assumes you
+    // are in this mode.
```

Comment:
The meaning of the last sentence is not clear to me. Do you actually mean:

TensorRT assume bilinear resize input tensor is in NCHW format?

Reply:
I meant that the input received by the converter is always NHWC, because that is enforced by TF.